lib/compress.js

```diff
@@ -338,7 +338,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant(compressor)) {
+                    if (!(var_decl.value instanceof AST_RegExp) && var_decl.value.is_constant(compressor)) {
```

Comment:
It's also used elsewhere in the code where an AST_RegExp literal is an acceptable AST_Constant. 

Although AST_RegExp is derived from AST_Constant it's more of a pseudo constant.